### PR TITLE
Kddimitrov/watch app

### PR DESCRIPTION
**Pull request by KristianDD**
## PR Checklist

- [ ] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [ ] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [ ] You have signed the [CLA](http://www.nativescript.org/cla).
- [ ] All existing tests are passing: https://github.com/NativeScript/nativescript-cli/blob/master/CONTRIBUTING.md#contribute-to-the-code-base
- [ ] Tests for the changes are included.

## What is the new behavior?

Fixes/Implements/Closes https://github.com/NativeScript/nativescript-cli/issues/4541

### CODE REVIEW

**Comment by KristianDD** [comment]
test cli-smoke

**Comment by KristianDD** [comment]
test cli-smoke

File: lib/services/ios-watch-app-service.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+		const identifierParts = identifier.split(".");
+		identifierParts.pop();
+		const wkAppBundleIdentifier = identifierParts.join(".");
+
+		this.setXcodeTargetBuildConfigurationProperties([
+			{name: "PRODUCT_BUNDLE_IDENTIFIER", value: identifier},
+			{name: "SDKROOT", value: "watchos"},
+			{name: "TARGETED_DEVICE_FAMILY", value: IOSDeviceTargets.watchos},
+			{name: "WATCHOS_DEPLOYMENT_TARGET", value: 5.2},
```
Comment (rosen-vladimirov) [inline lib/services/ios-watch-app-service.ts]: should we hardcode this?

Comment (KristianDD) [reply lib/services/ios-watch-app-service.ts]: I agree that its not obvious, but I left a way for the user to override this setting in the `watchapp.json` and `extension.json`. The method call [here ](https://github.com/NativeScript/nativescript-cli/pull/4568/files#diff-6a6e27b65c72d3aa7c4350bd602c3a1cR79) will read the JSON file and set all target build configuration properties listed in the property - the code is [here](https://github.com/NativeScript/nativescript-cli/pull/4568/files#diff-609e22cb6bb3cad20d7ba450784b4184R89). 

For an example how the JSON file will look like you can check point 7 in the related issue description [here](https://github.com/NativeScript/nativescript-cli/issues/4541).

Comment (rosen-vladimirov) [reply lib/services/ios-watch-app-service.ts]: Thanks for the great explanation and please excuse me for not checking the related issue.

**Pull request merged**